services/brig/src/Brig/User/Auth.hs

```diff
@@ -329,9 +331,12 @@ newAccess uid cid ct cl = do
       t <- lift $ newAccessToken @u @a ck Nothing
       pure $ Access t (Just ck)
 
+-- FUTUREWORK(mangoiv): unfortunately this uses lookupHandleImple explicity, in future, this
+-- should use lookuHandle but should be moved into some place where this resolveLoginId is done
+-- abstractly, perhaps some authentication subsystem
 resolveLoginId :: (MonadClient m, MonadReader Env m) => LoginId -> ExceptT LoginError m UserId
 resolveLoginId li = do
-  usr <- validateLoginId li >>= lift . either lookupKey lookupHandle
+  usr <- validateLoginId li >>= lift . either lookupKey (liftClient . lookupHandleImpl LocalQuorum)
```

Comment:
Why can't we use the `liftSem` here to call the subsystem action? I think its fine if we write it in terms of `AppT` instead of the constraints.

Reply:
45ddd8d7821b7609bb51717cd52fb82d326283d5 ?